packages/core/src/config/config.ts

```diff
@@ -0,0 +1,166 @@
+import { ConfigUrl } from '../index.js';
+import type {
+  Domain,
+  Domainlike,
+  EthereumConfig,
+  Network,
+  Resource,
+  SubstrateConfig,
+  SygmaConfig,
+} from '../types.js';
+import { Environment } from '../types.js';
+
+import { localConfig } from './localConfig.js';
+
+export class Config {
+  /**
+   * Cache storing all sygma
+   * environment configurations
+   */
+  configuration!: SygmaConfig;
+  initialized = false;
+  /**
+   * Initialize and store all
+   * sygma configurations
+   */
+  async init(environment: Environment): Promise<void> {
+    if (environment === Environment.LOCAL) {
+      this.configuration = localConfig;
+    } else {
+      const response = await fetch(this.getConfigUrl(environment));
+      const config = (await response.json()) as SygmaConfig;
+      this.configuration = config;
+      // initialized is set to true when
+      // all configurations have been fetched
+      this.initialized = true;
```

Comment:
Should we put it outside if/else condition as even if we use localConfig we still need to make initialized as "true"?